packages/app/src/cli/models/extensions/specifications/app_config_webhook.test.ts

```diff
@@ -246,5 +279,62 @@ describe('webhooks', () => {
         },
       })
     })
+    test('when subscriptions share the application_url base, simplify with a relative path', () => {
+      // Given
+      const object = {
+        api_version: '2021-01',
+        subscriptions: [
+          {
+            topic: 'products/update',
+            uri: 'https://my-app-url.com/products',
+          },
+          {
+            topic: 'products/delete',
+            uri: 'https://my-app-url.com/products',
+          },
+          {
+            topic: 'orders/update',
+            uri: 'https://my-app-url.com/orders',
+          },
+          {
+            topic: 'customers/create',
+            uri: 'https://customers-url.com',
+          },
+          {
+            topic: 'customers/delete',
+            uri: 'pubsub://absolute-feat-test:pub-sub-topic',
+          },
+        ],
+      }
+      const webhookSpec = spec
+
+      // When
+      const result = webhookSpec.reverseTransform!(object, {application_url: 'https://my-app-url.com'})
```

Comment:
what happens when if the `application_url` is invalid?

Reply:
I'm not sure if this should be tested in `loader_test` or this test file

Reply:
Why should this be tested in loader? The essence of those tests are more integration tests. These are simply unit tests, hence why we directly call `webhookSpec.reverseTransform`. We don't really care about anything that happens above this point in the code, we can assume our other test coverage handles that (which it does as you caught in your previous comment!). We only care about testing the logic contained in this transform function.

Reply:
I suggested `loader_test` because it's where I saw we were testing for errors after a deploy occurs on the CLI (so yes integration tests).  The tests you linked to there was testing specifically for `uri` field, but not for errors which might be thrown if the `application_url` field is invalid (ie. for example, developer starts out the TOML with relative path and `application_url`, or edits the `application_url` after the TOML is transformed)?